Ffweb 839 upgrade guide (#163)
FACT-Finder-Web-Components/documentation: markdown/3.0/en/documentation/upgrade-guide.md

In general you can keep track of our progress, changes and new features in the [release notes](https://github.com/FACT-Finder-Web-Components/ff-web-components/blob/master/CHANGELOG.md).

> The main difference of _1.2.x_ and _3.0.0_ is, that we have internal moved from [Polymer 1](https://polymer-library.polymer-project.org/1.0/docs/devguide/feature-overview) to [Polymer 3](https://polymer-library.polymer-project.org/3.0/docs/about_30) and are half way through to migrating to [LitElement](https://lit-element.polymer-project.org/).

Reviewer: ```suggestion
The main difference of _1.2.x_ and _3.0.0_ is, that we internally moved from [Polymer 1](https://polymer-library.polymer-project.org/1.0/docs/devguide/feature-overview) to [Polymer 3](https://polymer-library.polymer-project.org/3.0/docs/about_30) and are half way done migrating to [LitElement](https://lit-element.polymer-project.org/).
```

There are only two major changes and some minor API changes to take care of when upgrading from version 1.2.x to version 3.0.0. In addition, CSS mixins have been removed. You can use normal CSS selector instead. Further more `ffw-` is introduced as the prefix for FACT-Finder Web Components custom class names.

### 2 major changes

> - Aligning with the current trend FACT-Finder Web Components are now shipped as plain JavaScript instead of HTML Imports. Which is why you have to load `bundle.js` instead of the previous HTML import.

Reviewer: ```suggestion
- As [HTML-Imports](https://w3c.github.io/webcomponents/spec/imports/) didn't become part of the HTML and DOM specs, unlike the rest of the Web Components spec, FACT-Finder Web Components now ship as plain JavaScript instead of HTML. Which is why you have to load `bundle.js` instead of the previous HTML-Imports.
```

Reviewer 2: Merged

ff-nav-element a
```

#### ff-navigation-item
```
--container-mixin
ff-navigation-item .ffw-item-container
```

#### ff-onfocus-suggest
```
--suggest-container
ff-onfocus-suggest .ffw-suggestContainer

--suggest-container-wrapper
ff-onfocus-suggest .ffw-suggestContainerWrapper
```

#### ff-paging-dropdown
```
--dropdown-item-container
ff-paging-dropdown .ffw-paging-dropdown-container
```

#### ff-paging-item
```
--paging-item-cursor-mixin
ff-paging-item .ffw-cursor
```

#### ff-products-per-page-dropdown
```

> `--dropdown-item-container`

Reviewer 2: ```suggestion
--dropdown-item-container
```

> .ffw-ppp-dropdown-container

Reviewer 2: ```suggestion
ff-products-per-page-dropdown .ffw-ppp-dropdown-container
```

```

#### ff-sortbox
```

> `--sort-item-container`

Reviewer 2: ```suggestion
--sort-item-container
```

> .ffw-dropdown-container

Reviewer 2: ```suggestion
ff-sortbox .ffw-dropdown-container
```
